curvefs/src/metaserver/trash.cpp

```diff
@@ -56,75 +56,99 @@ void TrashImpl::Init(const TrashOption &option) {
     mdsClient_ = option.mdsClient;
     copysetNode_ = copyset::CopysetNodeManager::GetInstance().
         GetSharedCopysetNode(poolId_, copysetId_);
+
     isStop_ = false;
 }
 
 void TrashImpl::StopScan() {
     isStop_ = true;
 }
 
-bool TrashImpl::IsStop() {
-    return isStop_;
-}
-
-void TrashImpl::Add(uint64_t inodeId, uint64_t dtime) {
+void TrashImpl::Add(uint64_t inodeId, uint64_t dtime, bool deleted) {
     if (isStop_) {
         return;
     }
+    TrashItem item;
+    item.inodeId = inodeId;
+    item.dtime = dtime;
+    item.deleted = deleted;
     LockGuard lg(itemsMutex_);
-    trashItems_[inodeId] = dtime;
-    VLOG(6) << "Add Trash Item success, fsId = " << fsId_
-            << ", partitionId = " << partitionId_
-            << ", inodeId = " << inodeId
-            << ", dtime = " << dtime;
+    trashItems_.push_back(item);
+    VLOG(6) << "Add trash item success"
+            << ", item.inodeId = " << item.inodeId
+            << ", item.dtime = " << item.dtime
+            << ", item.deleted = " << item.deleted;
+}
+
+bool TrashImpl::IsStop() {
+    return isStop_;
 }
 
 void TrashImpl::Remove(uint64_t inodeId) {
     if (isStop_) {
         return;
     }
+    TrashItem item;
+    item.inodeId = inodeId;
     LockGuard lg(itemsMutex_);
-    trashItems_.erase(inodeId);
+    auto it = std::find(trashItems_.begin(), trashItems_.end(), item);
+    if (it != trashItems_.end()) {
```

Comment:
The remove of rocksdb data can do here also.

Reply:
fix